### PR TITLE
fix: ortto remove phn:phone field if no phone is there in payload

### PR DESCRIPTION
**Pull request by anantjain45823**
## What are the changes introduced in this PR?
When there is no phone field provided we were previously sending `phn:phone ={}` which was resulting in error as ` empty phone value for phn::phone"`.
In this PR we have put a check that if no phone field is provided we won't map it and there won't be any phn:phone field getting delivered to ortto.

## What is the related Linear task?

Resolves INT-1293

## Please explain the objectives of your changes below
This change will result in successful delivery of jobs when there is no phone provided in the payload

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

We won't be mapping `phn:phone `field even there is nothing to map

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [x] All related docs linked with the PR?

- [x] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [x] Is the type of change in the PR title appropriate as per the changes?

- [x] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

File: src/cdk/v2/destinations/ortto/procWorkflow.yaml [hunk 1]
```diff
@@ -46,7 +47,7 @@ steps:
           "dtz::b": $.getBirthdayObj({{{{$.getGenericPaths("birthday")}}}}),
           "str::ei": {{{{$.getGenericPaths("userId")}}}},
           "str::language": .context.traits.language || .context.locale,
-          "phn::phone": {"n": {{{{$.getGenericPaths("phone")}}}}},
+          "phn::phone": phone ? {"n": phone} : undefined,
```
Comment (koladilip) [inline src/cdk/v2/destinations/ortto/procWorkflow.yaml:50]: you can skip `: undefined` and write just `phone ? {"n": phone}`

File: src/cdk/v2/destinations/ortto/procWorkflow.yaml [hunk 2]
```diff
@@ -57,6 +58,7 @@ steps:
               const trimmedOrttoAttribute = attribute.orttoAttribute.trim().toLowerCase().replace(new RegExp('\\s+', 'g'),'-');
               commonFields.fields[$.fieldTypeMap[attribute.type]+":cm:"+trimmedOrttoAttribute] = $.originalInput.message.context.traits[attribute.rudderTraits]
             )[]
+      console.log("ABVC",commonFields.fields["phn::phone"])
```
Comment (koladilip) [inline src/cdk/v2/destinations/ortto/procWorkflow.yaml:61]: remove this

File: src/cdk/v2/destinations/ortto/procWorkflow.yaml [hunk 3]
```diff
@@ -57,6 +58,7 @@ steps:
...
       commonFields.fields = $.removeUndefinedAndNullValues(commonFields.fields)
```
Comment (koladilip) [inline src/cdk/v2/destinations/ortto/procWorkflow.yaml:62]: commonFields.fields = $.removeUndefinedAndNullValues(commonFields.fields);

=> is this required as `$.removeUndefinedAndNullValues(commonFields)` should take care of this right?

Comment (anantjain45823) [reply src/cdk/v2/destinations/ortto/procWorkflow.yaml:62]: Nope it is not required put this for debugging purpose. Removed

File: src/cdk/v2/destinations/ortto/procWorkflow.yaml (hunk 2 above)
Comment (anantjain45823) [reply src/cdk/v2/destinations/ortto/procWorkflow.yaml:61]: Done

File: src/cdk/v2/destinations/ortto/procWorkflow.yaml (hunk 1 above)
Comment (anantjain45823) [reply src/cdk/v2/destinations/ortto/procWorkflow.yaml:50]: Done

File: src/cdk/v2/destinations/ortto/procWorkflow.yaml (hunk 3 above)
Comment (anantjain45823) [reply src/cdk/v2/destinations/ortto/procWorkflow.yaml]: `$.removeUndefinedAndNullValues(commonFields)` does not take care of that as it is not a recursive function

**Review by koladilip — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**